### DISCUSSION TITLE
Does ArgoCD have a timeout, or rely only on the degraded state

### DISCUSSION THREAD

**Discussion by UndoPy** [argoproj/argo-cd#5600] [Q&A] [answered]
I have a CRD that I expect to take a long time to come up and be healthy. I've configured a lua script to take the statuses I'm sending but currently I often get a failed sync status and I'm wondering if there's a configurable timeout, or if I'm sending degraded state in error

**Comment by jgwest** [comment, marked as answer]
Hi @UndoPy , you might try increasing the `--app-resync` period of the Argo CD Application Controller... my guess is that a new resync begins before your previous sync has become healthy, due to the long CRD come up time. More information on app-resync is available on the [High Availability](https://argoproj.github.io/argo-cd/operator-manual/high_availability/) doc page.
